src/zenml/integrations/github/code_repositories/github_code_repository.py

```diff
@@ -202,3 +215,55 @@ def check_remote_url(self, url: str) -> bool:
             return True
 
         return False
+
+
+def create_symlink_in_local_repo_copy(
```

Comment:
I think it would be nice to have a check whether the symlink target is actually part of the repository contents. In other cases, I feel like we should not create the symlink at all.

Reply:
What do you mean exactly? 
Currently the function is only called when a symlink is found in the repo content.

Reply:
What I meant here is: The symlink source could be a path which is outside of the contents of the repository, for example like this:

```bash
echo outside > outside.txt
mkdir repo
cd repo
git init
ln -s ../outside.txt outside_symlink
```

In most cases, when downloading the symlink the source will not exist, but in some other cases it might lead to unexpected behaviour -> I would suggest we just prevent this case alltogether and don't try to create a symlink if the file is outside of the directory we download. What do you think?